network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -81,7 +142,18 @@ public void registerExecutor(
       ExecutorShuffleInfo executorInfo) {
     AppExecId fullId = new AppExecId(appId, execId);
     logger.info("Registered executor {} with {}", fullId, executorInfo);
-    executors.put(fullId, executorInfo);
+    synchronized (executors) {
+      executors.put(fullId, executorInfo);
+      try {
+        if (db != null) {
+          byte[] key = dbAppExecKey(new AppExecId(appId, execId));
```

Comment:
We already have an AppExecId above on line 143 (fullIId) we should be able to use that here